### PR TITLE
Update the e2e tests README

### PR DESCRIPTION
**Pull request by davehunt**
Updates and cleanup of the e2e tests README.

@chartjes r? /cc @stephendonner @m8ttyB

### CODE REVIEW

File: e2e-tests/README.md [hunk 1]
```diff
@@ -48,34 +48,35 @@ ___Running tests on SauceLabs___
 
 To use SauceLabs instead of an instance of Firefox running locally, do the following:
 
-Create a text file called `.saucelabs` and put it in the e2e-tests directory. Get a username and key for SauceLabs and then add the following details to the `.saucelabs` file:
+Create a text file called `.saucelabs` and put it in your home directory. Get a
```
Comment (davehunt) [inline e2e-tests/README.md]: It's best for this to be outside of the repository so it's not accidentally included in a pull request.

File: e2e-tests/README.md [hunk 2]
```diff
@@ -48,34 +48,35 @@ ___Running tests on SauceLabs___
...
 	$ tox -e py27 -- --driver SauceLabs --capability browserName Firefox
 
-If you wish to run them against different environemts, set `PYTEST_BASE_URL` as indicated in the sections above for running tests against localhost or production
+If you wish to run them against different environments, set `PYTEST_BASE_URL`
+as indicated in the sections above for running tests against localhost or
+production.
 
 ___Running tests using headless Firefox___
 
-NOTE: These instructions are for Linux or macOS only.
```
Comment (davehunt) [inline e2e-tests/README.md]: Headless should work on all environments.

File: e2e-tests/README.md [hunk 3]
```diff
@@ -48,34 +48,35 @@ ___Running tests on SauceLabs___
...
-Set an environment variable `MOZ_HEADLESS` to be '1'. Check the documentation for your shell on how to do this. For zsh you can do using the following command:
+Set an environment variable `MOZ_HEADLESS` to be '1'. Check the documentation
+for your shell on how to do this. For zsh you can do using the following
+command:
 
 	$ export MOZ_HEADLESS=1
 
-Next you need to set an environment variable `MOZ_BINARY_PATH` to be the location of the binary of Firefox you want to use. For example, if you were using zsh and Firefox Nightly (which supports running in headless mode) on macOS:
-
-        $ export MOZ_BINARY_PATH='/Applications/FirefoxNightly.app/Contents/MacOS/firefox'
```
Comment (davehunt) [inline e2e-tests/README.md]: This is not a requirement for headless mode, and we already cover specifying a Firefox binary path earlier in the README.

File: e2e-tests/README.md [hunk 4]
```diff
@@ -109,12 +110,9 @@ Output of a test run should look something like this:
 
     ============== 63 passed, 10 xfailed in 970.27 seconds ===============
 
-__Note__
-"~" will not resolve to the home directory when used in the py.test command line.
```
Comment (davehunt) [inline e2e-tests/README.md]: I tested this and it is not true, at least for test paths. If anyone is able to replicate this please file a bug and mention me. We should fix it.

File: e2e-tests/README.md [hunk 5]
```diff
@@ -128,60 +126,15 @@ Tips and tricks
...
-[pytest-selenium] provides the ability to run tests against [many other][test envs] browser environments -- consider using a different driver executable or external provider.
-
-It is important to note that tests must pass with Firefox driver otherwise they will not be accepted for merging.
-
-    $ # tests must pass with Firefox driver before submitting a pull request
-    $ py.test --driver PhantomJS --driver-path `which phantomjs` ...
-
-__xvfb__
-
-Xvfb provides a fairly easily work around on Linux.
```
Comment (davehunt) [inline e2e-tests/README.md]: No need for this now we have headless mode.

File: e2e-tests/README.md [hunk 6]
```diff
@@ -128,60 +126,15 @@ Tips and tricks
...
-
-        $ xvfb-run py.test ...
-
-
-    This creates a virtual X session for Firefox to run in, and sets
-    up all the fiddly environment variables to get this working
-    well. The tests will run as normal, and no windows will open, if
-    all is working right.
-
-License
```
Comment (davehunt) [inline e2e-tests/README.md]: This is in the top level so we don't need to repeat it here.

File: e2e-tests/README.md [hunk 7]
```diff
@@ -128,60 +126,15 @@ Tips and tricks
...
-[fxtest]:http://02.chat.mibbit.com/?server=irc.mozilla.org&channel=#fx-test
-[GitWin]: http://help.github.com/win-set-up-git/
-[GitMacOSX]: http://help.github.com/mac-set-up-git/
-[GitLinux]: http://help.github.com/linux-set-up-git/
-[breakpad]:http://02.chat.mibbit.com/?server=irc.mozilla.org&channel=#breakpad
-[venv]: http://pypi.python.org/pypi/virtualenv
-[wrapper]: http://www.doughellmann.com/projects/virtualenvwrapper/
-[GitHub Templates]: https://github.com/mozilla/mozwebqa-examples
-[Style Guide]: https://wiki.mozilla.org/QA/Execution/Web_Testing/Docs/Automation/StyleGuide
-[MPL]: http://www.mozilla.org/MPL/2.0/
```
Comment (davehunt) [inline e2e-tests/README.md]: Lots of unused links.

**Comment by willkg** [comment]
Everything here looks good to me. @chartjes is on PTO--not sure if he's checking his email. I'll wait until the end of the day and if he hasn't r+d by then, I'll r+ it and land it.

Thank you for doing this!

**Comment by chartjes** [comment]
+1

**Comment by willkg** [comment]
Thank you!

**Pull request merged**